### PR TITLE
LYNX-334: Add doc for place order changes and new queries

### PR DESCRIPTION
**Pull request by eliseacornejo**
## Purpose of this pull request

This pull request (PR) is to add two new queries that are **guestOrder** and **guestOrderByToken**. Also added some changes on the **placeOrder** documentation. 
It is related to this ticket: [LYNX-334](https://jira.corp.adobe.com/browse/LYNX-334)

## Affected pages

- https://developer.adobe.com/commerce/webapi/graphql/schema/cart/mutations/place-order/
- https://developer.adobe.com/commerce/webapi/graphql/schema/orders/queries/guest-order/ (new page)
- https://developer.adobe.com/commerce/webapi/graphql/schema/orders/queries/guest-order-by-token/ (new page)

## Links to Magento Open Source code

- ...

### CODE REVIEW

File: src/pages/graphql/schema/cart/mutations/place-order.md [hunk 1]
```diff
@@ -47,8 +47,12 @@ mutation {
       cart_id: "IeTUiU0oCXjm0uRqGCOuhQ2AuQatogjG"
     }
   ) {
-    order {
-      order_number
+    orderV2 {
```
Comment (keharper) [inline src/pages/graphql/schema/cart/mutations/place-order.md:50]: Add a paragraph immediately above the InlineArt in the description section stating that as of 2.4.7, the PlaceOrderOutput object can contain orderV2, which contains full details about the order.

File: src/pages/graphql/schema/orders/queries/guest-order.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+---
+title: guestOrder query
+---
+
+# guestOrder query
+
+Use the `guestOrder` query to retrieve guest order details based on number, email and postcode.
```
Comment (keharper) [inline src/pages/graphql/schema/orders/queries/guest-order.md:7]:
```suggestion
Use the `guestOrder` query to retrieve details about an order placed by a guest or customer who is not logged in. To retrieve this information, the guest must supply the order number, email, and postal code used to create the order.
```

**Review by keharper — CHANGES_REQUESTED** [review]

File: src/pages/graphql/schema/orders/queries/guest-order.md (hunk 2 above)
Comment (eliseacornejo) [reply src/pages/graphql/schema/orders/queries/guest-order.md]: Done!

File: src/pages/graphql/schema/cart/mutations/place-order.md (hunk 1 above)
Comment (eliseacornejo) [reply src/pages/graphql/schema/cart/mutations/place-order.md:52]: Done!

File: src/pages/graphql/schema/orders/queries/guest-order-by-token.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+---
+title: guestOrderByToken query
+---
+
+# guestOrderByToken query
+
+Use the `guestOrderByToken` query to retrieve guest order details based on token.
```
[the remark this answers is not in the document]
Comment (eliseacornejo) [reply src/pages/graphql/schema/orders/queries/guest-order-by-token.md:7]: The token comes from the order, it is token = encrypted(number + email + postcode)

[the remark this answers is not in the document]
Comment (keharper) [reply src/pages/graphql/schema/orders/queries/guest-order-by-token.md:7]: @eliseacornejo I'm asking on behalf of the external user. Please provide guidance on how to generate a token value. Maybe a bit of PHP code (or other language).

[the remark this answers is not in the document]
Comment (eliseacornejo) [reply src/pages/graphql/schema/orders/queries/guest-order-by-token.md:7]: You can get the token from the placeOrder mutation response. I added it on the documentation. This is not something that you can generate outside.

File: src/pages/graphql/schema/cart/mutations/place-order.md [hunk 4]
```diff
@@ -16,6 +16,8 @@ Perform the following actions before using the `placeOrder` mutation:
 -  Set the payment method
 -  For guest customers, assign an email to the cart
 
+As of Commerce 2.4.7, the PlaceOrderOutput object can contain orderV2, which contains full details about the order.
```
Comment (keharper) [inline src/pages/graphql/schema/cart/mutations/place-order.md:19]:
```suggestion
As of Commerce 2.4.7, the `PlaceOrderOutput` object can contain `orderV2`, which contains full details about the order. When the mutation runs, Commerce generate a token in the `orderV2.token` field. You can supply the token value to the [`guestOrderByToken` query](../../orders/queries/guest-order-by-token.md) to retrieve details about an order created by a guest shopper.
```

File: src/pages/graphql/schema/orders/queries/guest-order-by-token.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+You can retrieve the token from the **CustomerOrder** object on the [placeOrder mutation](../../cart/mutations/place-order.md) response.
```
Comment (keharper) [inline src/pages/graphql/schema/orders/queries/guest-order-by-token.md:8]:
```suggestion
You can retrieve the token from the `CustomerOrder` object on the [`placeOrder` mutation](../../cart/mutations/place-order.md) response.
```

File: src/pages/graphql/schema/orders/queries/guest-order.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+## Example usage
+
+**Request:**
+
+```graphql
+{
+  guestOrder(
+    input: {
+      number: "000000001",
+      email: "test@test.com",
```
Comment (keharper) [inline src/pages/graphql/schema/orders/queries/guest-order.md:26]:
```suggestion
      email: "test@example.com",
```

File: src/pages/graphql/schema/cart/mutations/place-order.md (hunk 4 above)
Comment (eliseacornejo) [reply src/pages/graphql/schema/cart/mutations/place-order.md]: Done

File: src/pages/graphql/schema/orders/queries/guest-order-by-token.md (hunk 5 above)
Comment (eliseacornejo) [reply src/pages/graphql/schema/orders/queries/guest-order-by-token.md]: Done

File: src/pages/graphql/schema/orders/queries/guest-order.md (hunk 6 above)
Comment (eliseacornejo) [reply src/pages/graphql/schema/orders/queries/guest-order.md]: Done

**Review by keharper — APPROVED** [review]

**Pull request merged**